### PR TITLE
Initial setup of VPN Gateway for AWS VPCs

### PR DESCRIPTION
**Pull request by miguelaferreira**
I saw that this repo was empty and since I've made some terraform configuration for connecting a VPN Gateway to a vpc (build with the VPC module in the registry), I thought of contributing it here.

I'll be happy to receive comments and improve this PR.

### CODE REVIEW

**Comment by antonbabenko** [comment]
Great initiative! 👍 

This module creates some `aws_vpn_*` resources as well as [terraform-aws-modules/vpc](https://github.com/terraform-aws-modules/terraform-aws-vpc) does, so it would be a good idea to specify the following:
1. What types of resources VPC module create in regards to VPN (it creates `aws_vpn_gateway` itself, but not VPN connection)
1. What types of resources this module create (mention it as a list in README close to the top as similar to [this](https://github.com/terraform-aws-modules/terraform-aws-security-group/blob/master/README.md))
1. Show small code snippets showing how to use only this module (you already have this) and another one showing how to combine this with VPC module.
1. Make creation of `aws_vpn_connection` optional, because somebody may want to do this outside of this scope (btw, edit `intralot_gateway_id`).
1. [aws_vpn_connection](https://www.terraform.io/docs/providers/aws/r/vpn_connection.html) supports few more arguments. Make sure that those are configurable with valid defaults.

Few general comments to this:
1. Add description to each `variable`
1. Add `outputs.tf` to include all possible attributes (there was at the very least [this](https://github.com/terraform-community-modules/tf_aws_vpn_gw/blob/master/outputs.tf)).

**Comment by miguelaferreira** [comment]
Thanks for the feedback, I will review it and update the PR accordingly.

**Comment by miguelaferreira** [comment]
@antonbabenko I've updated the PR can you check it again?

You asked for a minimum set of outputs that included the VPN Gateway id, but that's actually an input to the module, so I didn't add it as an output.

File: variables.tf [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+variable "vpc_subnet_route_table_ids" {
+  description = "The ids of the VPC subnets for which routes from the VPN Gateway will be propagated"
+  type        = "list"
+  default     = []
+}
+
+variable "tags" {
+  description = "List fo tags to be added to the VPN Connection resource (only if `create_vpn_connection = true`)"
+  type        = "list"
```
Comment (antonbabenko) [inline variables.tf]: `tags` should be type of `map`. [Read here](https://www.terraform.io/docs/configuration/interpolation.html#merge-map1-map2-)

**Comment by miguelaferreira** [comment]
I've fixed the tags variable type and added 3 examples (all using the VPC module) that I think cover all possibilities of usage of the module

**Comment by antonbabenko** [comment]
It seems there are some blocking issues with the Terraform registry, so that [v1.0.0](https://github.com/terraform-aws-modules/terraform-aws-vpn-gateway/releases/tag/v1.0.0) is not visible [there](https://registry.terraform.io/modules/terraform-aws-modules/vpn-gateway/aws/) in a drop-down list. It says that it may take a few minutes. I will check later.

**Comment by antonbabenko** [comment]
https://registry.terraform.io/modules/terraform-aws-modules/vpn-gateway/aws/1.0.0 - when the code contains validation error it doesn't appear in the registry. Problem fixed.

**Pull request merged**